packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -114,17 +115,18 @@ if (__DEV__) {
   const didWarnStateUpdateForUnmountedComponent = {};
 
   warnAboutUpdateOnUnmounted = function(fiber: Fiber) {
+    // We show the whole stack but dedupe on the top component's name because
```

Comment:
because...?

Reply:
because the problematic code practically always lies inside that component. :)